### DISCUSSION TITLE
GeneActivity based on genebody accessibility

### DISCUSSION THREAD

**Discussion by ZunpengLiu** [GreenleafLab/ArchR#1739] [General]
Hi,

Thanks for developing and managing this robust tool! 

I was wondering if I have a custom gene list and want to calculate the gene activity considering gene body accessibility only. So, I use the addGeneScoreMatrix function with the parameters "extendUpstream = c(0, 0),
  extendDownstream = c(0, 0),
  geneUpstream = 0,
  geneDownstream = 0".

Do you think the parameters can help me achieve my purpose? Do you have any suggestions about this?

Many thanks!
  
Zunpeng

**Comment by rcorces** [comment]
Minimally, you also need to set `useTSS = FALSE` otherwise `geneUpstream` and `geneDownstream` dont take effect. With that addition, I think you would accomplish what you are looking for.

**Reply by ZunpengLiu** [reply]
Thank you so much! 

Yes, I also set `useTSS = FALSE`. I may not fully understand the meaning of `extendUpstream` and `extendDownstream` . I wonder if it is reasonable to set them as `c(0,0)`.

**Reply by rcorces** [reply]
`extendUpstream` and `extendDownstream` will have no effect when `useTSS = FALSE`
